spec/lib/url_cache_spec.rb

```diff
@@ -24,44 +24,93 @@
     end
   end
 
+  shared_examples "memcached exceptions" do |dalli_method_name|
+    describe "memcached exception handling" do
+      let(:dalli_client) { instance_double(Dalli::Client) }
+      let(:url_cache)    { described_class.new(cache: dalli_client) }
+
+      let(:error_message) { "An error has occured!" }
+      let(:dalli_error)   { dalli_error_class.new(error_message) }
+
+      before do
+        allow(dalli_client)
+          .to receive(dalli_method_name)
+          .and_raise(dalli_error)
+      end
+
+      context "when a Dalli::RingError is raised" do
+        let(:dalli_error_class) { Dalli::RingError }
+
+        it "should retry a few times" do
```

Comment:
refrain from using "should" also a few is pretty loose, I'd suggest "it retries #{expected_retries} times" and we defined that variable.

Reply:
(same with all "shoulds")

Reply:
:+1:

Reply:
Changed it in 4d665e0

All other specs (specs without an explicit description) outputs `should`, that's probably why I went with that word :)

```
Twingly::UrlCache
  should respond to #ttl
  should respond to #cache!
  should respond to #cached?
  #ttl
    ...
```

Reply:
yeah, I wish RSpec would change that :)